From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/578e5aea3cbb
From: xen-devel@lists.xenproject.org
Patch-mainline: n/a
Subject: scsifront: don't use bitfields for indicators modified under different locks

It is no good idea to have bitfields in the same byte which are modified
with different locks held (or, in this case, one with lock and one
without). Use at least bytes for this purpose.

Signed-off-by: Juergen Gross <jgross@suse.com>
Committed-by: Jan Beulich <jbeulich@suse.com>

--- sle11sp3.orig/drivers/xen/scsifront/common.h	2015-03-03 11:55:13.000000000 +0100
+++ sle11sp3/drivers/xen/scsifront/common.h	2015-02-18 14:52:58.000000000 +0100
@@ -113,8 +113,8 @@ struct vscsifrnt_info {
 	struct task_struct *kthread;
 	wait_queue_head_t wq;
 	wait_queue_head_t wq_sync;
-	unsigned int waiting_resp:1;
-	unsigned int waiting_sync:1;
+	unsigned char waiting_resp;
+	unsigned char waiting_sync;
 };
 
 #define DPRINTK(_f, _a...)				\
